File: src/service/ModelBuilder.Core/Models/Connector.cs
using System;
using Mb.Core.Enums;

namespace Mb.Core.Models
{
    public class Connector
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();
        public string Name { get; set; }
        public ConnectorType Type { get; set; }
        public TerminalCategory TerminalCategory { get; set; }
        public TerminalType TerminalType { get; set; }
        public RelationType RelationType { get; set; }

        public virtual string NodeId { get; set; }
        public virtual Node Node { get; set; }
    }
}

File: src/client/src/models/classes/Connector.ts
import { Attribute, ConnectorType, ConnectorVisibility, EnumBase, RelationType } from "..";

export const CONNECTOR_KIND = "Connector";

class Connector {
  id: string;
  iri: string;
  domain: string;
  name: string;
  type: ConnectorType;
  semanticReference: string;
  nodeId: string;
  nodeIri: string;
  connectorVisibility: ConnectorVisibility;
  isRequired: boolean;

  // Terminal
  color: string;
  terminalCategoryId: string;
  terminalCategory: EnumBase;
  attributes: Attribute[];
  terminalTypeId: string;

  // Only client
  inputOrder: number;
  outputOrder: number;

  // Relation
  relationType: RelationType;

  kind: string = CONNECTOR_KIND;

  constructor(connector: Connector) {
    Object.assign(this, connector);
  }
}

export default Connector;
